labrad/client.py

```diff
@@ -489,18 +490,52 @@ def port(self):
 
     @property
     def connected(self):
-        return self._backend.connected
+        return self._backend and self._backend.connected
 
     def connect(self, host, port=None, timeout=C.TIMEOUT, password=None,
                 tls_mode=C.MANAGER_TLS):
+        """(Re-)Establish a manager connection.
+
+        This is deprecated.  To establish a connection to a new manager,
+        call labrad.connect.  To establish a new connection to the existing
+        manager use Client.spawn().
+        """
         self._backend.connect(host, port=port, timeout=timeout,
                               password=password, tls_mode=tls_mode)
 
     def disconnect(self):
+        """Disconnect from the labrad manager.
+
+        This closes the connection for all Clients.
+        """
         self._backend.disconnect()
 
+    def close(self):
+        """Close this Client
+
+        Any other clients using the same connection will be unaffected.
+        This will expire all contexts allocated with the context() method.
+        It will not expire any contexts explicitly overriden by the user,
+        such as in a call to __init__ or Server.packet()
+        """
+        if self.connected:
+            for c in self._contexts:
+                self.manager.expire_context(context=c)
```

Comment:
@ejeffrey, I'm not sure we want to add this notion of individual client wrappers having their own lifetime and set of contexts. In any case, it seems to me this could be done separately from the backend thread safety changes.